### PR TITLE
Fix Prometheus exporter writing bool label values

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/4822
Design discussion issue #

## Changes

Prometheus exporter writes bool label values as lowercase `true` and `false` strings.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs [hunk 1]
```diff
@@ -222,10 +222,20 @@ public static int WriteLabel(byte[] buffer, int cursor, string labelKey, object
         buffer[cursor++] = unchecked((byte)'"');
 
         // In Prometheus, a label with an empty label value is considered equivalent to a label that does not exist.
-        cursor = WriteLabelValue(buffer, cursor, labelValue?.ToString() ?? string.Empty);
+        cursor = WriteLabelValue(buffer, cursor, GetLabelValueString(labelValue));
         buffer[cursor++] = unchecked((byte)'"');
 
         return cursor;
+
+        static string GetLabelValueString(object labelValue)
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs:230]: Consider adding a TODO comment here pointing to https://github.com/open-telemetry/opentelemetry-dotnet/issues/4822#issuecomment-1707328495.

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs [hunk 2]
```diff
@@ -222,10 +222,20 @@ public static int WriteLabel(byte[] buffer, int cursor, string labelKey, object
...
+        {
+            if (labelValue is bool b)
```
Comment (JamesNK) [inline src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs:232]:
```suggestion
            // TODO: Attribute values should be written as their JSON representation. Extra logic may need to be added here to correctly convert other .NET types.
            // More detail: https://github.com/open-telemetry/opentelemetry-dotnet/issues/4822#issuecomment-1707328495
            if (labelValue is bool b)
```

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs (hunk 1 above)
Comment (JamesNK) [reply src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs:230]: Done

**Review by cijothomas — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by utpilla — APPROVED** [review]
@JamesNK Could you add a CHANGELOG entry for this?

**Comment by JamesNK** [comment]
Done

**Pull request merged**